_universum/modules/vcs/perforce_vcs.py

```diff
@@ -561,8 +561,10 @@ def clean_workspace(self):
             self.p4.client = self.client_name
             report = self.p4.run_revert("//...")
             self.p4report(report)
-        except P4Exception:
-            pass
+        except P4Exception as e:
+            if "Client '{}' unknown".format(self.client_name) not in e.value \
+                    and "file(s) not opened on this client" not in e.value:
+                raise
```

Comment:
1. If we are using explicit try...except, we should use it for handling exception from delete_client as well. This would make this function more readable.
2. We should not do raise here. The exception will skip delete_client.